pkg/golinters/gci.go

```diff
@@ -15,33 +18,121 @@ import (
 const gciName = "gci"
 
 func NewGci(settings *config.GciSettings) *goanalysis.Linter {
-	var linterCfg map[string]map[string]interface{}
+	var mu sync.Mutex
+	var resIssues []goanalysis.Issue
+
+	analyzer := &analysis.Analyzer{
+		Name: gciName,
+		Doc:  goanalysis.TheOnlyanalyzerDoc,
+		Run:  goanalysis.DummyRun,
+	}
+
+	var cfg *gci.GciConfiguration
 	if settings != nil {
-		cfg := map[string]interface{}{
-			gci.NoInlineCommentsFlag:  settings.NoInlineComments,
-			gci.NoPrefixCommentsFlag:  settings.NoPrefixComments,
-			gci.SectionsFlag:          strings.Join(settings.Sections, gci.SectionDelimiter),
-			gci.SectionSeparatorsFlag: strings.Join(settings.SectionSeparator, gci.SectionDelimiter),
+		rawCfg := gci.GciStringConfiguration{
+			Cfg: gcicfg.FormatterConfiguration{
+				NoInlineComments: settings.NoInlineComments,
+				NoPrefixComments: settings.NoPrefixComments,
+			},
+			SectionStrings:          settings.Sections,
+			SectionSeparatorStrings: settings.SectionSeparator,
 		}
 
 		if settings.LocalPrefixes != "" {
```

Comment:
I want to remove LocalPrefixes here since I do not support that anymore in golangci-lint. @ldez

Reply:
inside golancci-lint we don't want to break